chill-java/src/main/java/com/twitter/chill/java/RegexSerializer.java

```diff
@@ -35,10 +35,11 @@ static public IKryoRegistrar registrar() {
     @Override
     public void write(Kryo kryo, Output output, Pattern pattern) {
         output.writeString(pattern.pattern());
+        output.writeInt(pattern.flags(), true);
     }
 
     @Override
     public Pattern read(Kryo kryo, Input input, Class<Pattern> patternClass) {
-        return Pattern.compile(input.readString());
+        return Pattern.compile(input.readString(), input.readInt(true));
     }
```

Comment:
sorry I left this sit so long, but it would be nice to add a test that shows the problem.